src/action-ros-ci.ts

```diff
@@ -199,56 +207,85 @@ async function installRosdeps(
  *
  * @param colconCommandPrefix the prefix to use before colcon commands
  * @param options the exec options
- * @param testPackageSelection the package selection option string
- * @param extra_options the extra options for 'colcon test'
- * @param coverageIgnorePattern the coverage filter pattern to use for lcov, or an empty string
+ * @param testPackageSelection the package selection option
+ * @param colconExtraArgs the extra args for 'colcon test'
+ * @param coverageIgnorePattern the coverage filter pattern to use for lcov
  */
 async function runTests(
-	colconCommandPrefix: string,
+	colconCommandPrefix: string[],
 	options: im.ExecOptions,
-	testPackageSelection: string,
-	extra_options: string[],
-	coverageIgnorePattern: string
+	testPackageSelection: string[],
+	colconExtraArgs: string[],
+	coverageIgnorePattern: string[]
 ): Promise<void> {
-	// ignoreReturnCode is set to true to avoid having a lack of coverage
-	// data fail the build.
-	const colconLcovInitialCmd = "colcon lcov-result --initial";
-	await execBashCommand(colconLcovInitialCmd, colconCommandPrefix, {
-		...options,
-		ignoreReturnCode: true,
-	});
+	if (!isWindows) {
```

Comment:
I know that much of this file is already built up in this way, but I think we should do better than `if windows; if !windows; if windows; if windows` - etc.

I'd prefer if we set up some more "meaningful" variables up front, then wrote a much less "aware" functionality below", e.g. pseudocode:

```
do_lcov_result = !isWindows;
if (isWindows) {
  colconExtraArgs += '--merge-install';
}
do_test = !isWindows;


if (do_lcov_result) {
  lcov_initial...
}
let colconTestCmd = [...extraArgs];
if (do_test) {
  call testCmd;
}
if (do_lcov_result) {
  lcov_final...;
}
```

Reply:
cleaned it up!